src/mixins/java/org/spongepowered/common/mixin/core/world/entity/EntityMixin.java

```diff
@@ -237,6 +236,27 @@
         this.impl$isConstructing = false;
     }
 
+    @Override
+    public boolean bridge$setPosition(final Vector3d position) {
+        if (this.removed) {
+            return false;
+        }
+
+        try (final CauseStackManager.StackFrame frame = PhaseTracker.getCauseStackManager().pushCauseFrame()) {
+            frame.pushCause(SpongeCommon.activePlugin());
```

Comment:
This should not be necessary as the plugin is set when we invoke a listener or scheduled task.